src/main/scala/com/normation/plugins/datasources/Repository.scala

```diff
@@ -199,8 +215,6 @@ class DataSourceRepoImpl(
 
   }
   private[this] def updateDataSourceScheduler(source: DataSource, delay: Option[FiniteDuration]): Unit = {
```

Comment:
So, the problem was most likely here: that method is not synchronized, and was used on non sync method. So what could have happened: 
[context $1]
- stop source.id $0, 
[switch context to $2]
- stop source.id $0, 
- create dss $2
- save dss $2
- start dss $2
[switch context $1]
- create dss $1
- save dss $1
- start dss $1

So now, both $1 and $2 run, but only $1 is know for rudder. This is bad.